scripts/filter_advisories.py

```diff
@@ -0,0 +1,126 @@
+import csv
+import json
+import re
+import sys
+from datetime import datetime
+from typing import Any, Dict, List
+
+import semantic_version
+
+
+def to_snake_case(s: str) -> str:
+    """Convert a string to snake_case."""
+    return re.sub(r"\W|^(?=\d)", "_", s).lower()
+
+
+def parse_advisories(filename: str) -> List[Dict[str, Any]]:
+    """Load advisories from a JSON file."""
+    with open(filename, "r") as f:
+        return json.load(f)
+
+
+def filter_advisories(
+    advisories: List[Dict[str, Any]],
+    minimal_version: semantic_version.Version,
+    last_run_date: datetime,
+) -> List[Dict[str, Any]]:
+    """Filter advisories based on the minimal vulnerable version and publication date."""
+    filtered = []
+    for advisory in advisories:
+        published_at = datetime.fromisoformat(advisory.get("published_at", ""))
+        if published_at > last_run_date:
+            for vulnerability in advisory.get("vulnerabilities", []):
+                vulnerable_range = vulnerability.get("vulnerable_version_range", "")
+                if vulnerable_range:
+                    try:
+                        if vulnerable_range == "ALL":
+                            filtered.append(advisory)
+                            break
+                        parsed_vulnerable_range = re.sub(
+                            "v(\\d[.])", "\\1", vulnerable_range
```

Comment:
minor thing but if you use `r` strings here you can avoid the double escaping, e.g. `r"v(\d[.])"`